aether-kernel/aether/kernel/api/views.py

```diff
@@ -310,6 +312,16 @@ class SubmissionViewSet(CustomViewSet):
     serializer_class = serializers.SubmissionSerializer
     filter_class = filters.SubmissionFilter
 
+    def get_queryset(self):
+        queryset = models.Submission.objects.all()
+        for k, v in self.request.query_params.items():
+            if k.startswith('payload__'):
+                try:
+                    kwargs = {k: json.loads(v)}
+                except json.decoder.JSONDecodeError:
+                    kwargs = {k: v}
+                queryset = queryset.filter(**kwargs)
+        return queryset
```

Comment:
Don't you prefer to include this in the `filters.SubmissionFilter` class as a method? :thinking:

Reply:
Maybe it's too much complicated :woman_shrugging:

Reply:
As far as I can tell, FilterSets (from the django filters library) do not accept completely dynamic query parameters, e.g. `payload__x__y__z` (we need this, since the models know nothing about the structure of the JSON payload.